### ISSUE TITLE
chunks broken in IE

### ISSUE THREAD

**Issue by jscheel** [webpack/webpack#2506]
I am trying to conditionally load a polyfill like this:

``` javascript
if (utils.needsHistoryPolyfill()) {
  require.ensure(['html5-history-api'], function(require) {
    require('html5-history-api');
    openApp();
  });
}
else {
  openApp();
}
```

If I do a hard-refresh, everything works. But on subsequent soft-refreshes (where the script jsonp response is loaded from cache in IE9), the chunk's code is never executed.

**Comment by jscheel** [comment]
This appears to be an issue with chunk loading and IE9 caching the ajax responses from the jsonp calls.

**Comment by jscheel** [comment]
I've updated the description to match what's happening.

**Comment by jscheel** [comment]
I'm thinking we could do something with checking the readyState to determine if a chunk is cached, then reloading it with a unique timestamp if the onload is fired, but the chunk is not properly loaded (which is what happens when IE does not actually evaluate the script that it loads).

**Comment by jscheel** [comment]
Interesting, at first I thought that it wasn't executing the script at all, but I can't reliably reproduce this issue. If that's not actually the case, we may be able to just resolve the promise instead.

**Comment by bang88** [comment]
Are the bug is fixed ?

**Comment by Markosyan-AR** [comment]
You can try to add some random to the scriptName parameter and push browser to load chunks manually
like:
$('head').append(<script scr="myChunk1?version=" + Math.Random()></script>)

**Comment by Phieo** [comment]
Are the bug is fixed?

**Comment by Markosyan-AR** [comment]
It isn't fixed yet :/

**Comment by Markosyan-AR** [comment]
looks like i've found the solution.
Compare to previous 1.0 version scripts are coming with async = true flag.
Tried to set it to false and it helps

I've branched from 2.1.0-beta.22 (latest versions doesnt works properly with my config) and there is fix: [https://github.com/Markosyan-AR/webpack/commit/8f0efcf7db0e22e9dfa9b300f50b0f1fc7d2d100](https://github.com/Markosyan-AR/webpack/commit/8f0efcf7db0e22e9dfa9b300f50b0f1fc7d2d100)

also faced my package.json to this branch
`"webpack": "https://github.com/Markosyan-AR/webpack/tarball/syncScriptsLoading"`

**Comment by NikitaEgorov** [comment]
I have the same problem too

**Comment by Markosyan-AR** [comment]
googled a good article about IE loading issues here:
http://www.guypo.com/ies-premature-execution-problem/
and solution is- set src attribute **after** adding element to the dom

**Comment by olaf89** [comment]
Had a same issue (IE10, webpack 2.1.0-beta.25), I have lazy loaded module, and it only works(additional chunk is loaded) first time i load a page, or after clearing browser cache. Looks like after content is cached chunk is just never loaded.

**Comment by mijay** [comment]
Had a same issue. Is there any progress?

**Comment by jdalton** [comment]
Patched and merged in #3551.

**Comment by TheLarkInn** [comment]
Yup thanks @jdalton closing thanks to #3551
